### PR TITLE
Fix (#7178): Follow-up fixes

### PR DESCRIPTION
**Pull request by OjusWiZard**
## Checklist

- [x] Add missing cache fields for settings in `/users` and `/history/debug`
- [x] Add support for `UpdateType`_versions in `DBSettings`
- [x] Fix bugs for not moving `UpdateType`_versions in DB upgrade

### CODE REVIEW

File: rotkehlchen/db/settings.py [hunk 1]
```diff
@@ -278,6 +289,8 @@ def db_settings_from_dict(
             specified_args[key] = int(value)
         elif key in STRING_KEYS:
             specified_args[key] = str(value)
+        elif key in UPDATE_TYPES:
+            specified_args[key] = value
```
Comment (yabirgb) [inline rotkehlchen/db/settings.py:293]: this will never return None right? so maybe return int(value)?

File: rotkehlchen/db/updates.py [hunk 2]
```diff
@@ -372,7 +372,7 @@ def check_for_updates(self, updates: Sequence[UpdateType] = tuple(UpdateType)) -
     def _check_for_last_version(cursor: 'DBCursor', update_type: UpdateType) -> int:
         """This method checks the database for the last local version of `update_type`"""
         found_version = cursor.execute(
-            'SELECT value FROM key_value_cache WHERE name=?',
+            'SELECT value FROM settings WHERE name=?',
```
Comment (yabirgb) [inline rotkehlchen/db/updates.py:375]: why this change? Weren't they suppose to be in the value cache table?

Comment (OjusWiZard) [reply rotkehlchen/db/updates.py:375]: We kept them in the settings table in the last PR https://github.com/rotki/rotki/pull/7178/#discussion_r1438117221

Comment (LefterisJP) [inline rotkehlchen/db/updates.py:375]: This is really bad that the change was not detected. Why did the tests here not fail before? 

Can you add a test that uses this so that this thing actually works and is covered?

File: rotkehlchen/tests/api/test_settings.py [hunk 3]
```diff
@@ -100,7 +101,8 @@ def test_querying_settings(rotkehlchen_api_server, username):
     assert json_data['message'] == ''
     assert result['version'] == ROTKEHLCHEN_DB_VERSION
     for setting in dataclasses.fields(DBSettings):
-        assert setting.name in result
+        if setting.name not in UPDATE_TYPES:
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_settings.py:104]: why do this? Makes no sense to inroduce this change

**Review by LefterisJP — APPROVED** [review]
LGTM
